test/integration/helpers/integration-test-editor.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import {
+	BlockEditorKeyboardShortcuts,
+	BlockEditorProvider,
+	BlockList,
+	BlockTools,
+	BlockInspector,
+	WritingFlow,
+	ObserveTyping,
+} from '@wordpress/block-editor';
+import { Popover, SlotFillProvider } from '@wordpress/components';
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { ShortcutProvider } from '@wordpress/keyboard-shortcuts';
+import '@wordpress/format-library';
+import {
+	createBlock,
+	unregisterBlockType,
+	getBlockTypes,
+} from '@wordpress/blocks';
+
+export function registerAllCoreBlocks() {
+	registerCoreBlocks();
+}
+
+export function unRegisterAllBlocks() {
+	getBlockTypes().forEach( ( { name } ) => unregisterBlockType( name ) );
+}
+
+export function createTestBlock( name, attributes ) {
+	return createBlock( name, attributes || {} );
+}
+
+/**
+ * Internal dependencies
+ */
+import { waitForStoreResolvers } from './wait-for-store-resolvers';
+
+export function Editor( { testBlocks } ) {
+	const [ blocks, updateBlocks ] = useState( [] );
```

Comment:
Should the initialState be `testBlocks`?

```suggestion
	const [ blocks, updateBlocks ] = useState( testBlocks );
```

Reply:
I read somewhere that it is considered an anti-pattern to use props for initial state, but it probably doesn't matter in this testing context so have updated.

Reply:
> It is considered an anti-pattern to use props for initial state

I'm curious as to why it's considered an anti-pattern? The initialState should only be set once so any updates to props won't trigger updates.

Reply:
yeh, I don't think it is an antipattern in that context https://vhudyma-blog.eu/react-antipatterns-props-in-initial-state/